code/game/machinery/computer/communications.dm

```diff
@@ -500,6 +501,418 @@
+	icon_state = "comm"
+	req_access = list(ACCESS_MARINE_CMO)
+	circuit = "/obj/item/circuitboard/computer/communications"
+	prints_intercept = FALSE
+	authenticated = 0
+	list/messagetitle = list()
+	list/messagetext = list()
+	currmsg = 0
+	aicurrmsg = 0
+	state = STATE_DEFAULT
+	aistate = STATE_DEFAULT
+	cooldown_message = FALSE //Based on world.time.
+	cooldown_request = FALSE
+	cooldown_central = FALSE
+	just_called = FALSE
+	tmp_alertlevel = SEC_LEVEL_GREEN
+
+	status_display_freq = "1435"
+	stat_msg1
+	stat_msg2
+
+	datum/announcement/priority/command/crew_announcement = new
+
+/obj/machinery/computer/communications/cmo/Topic(href, href_list)
+	. = ..()
+	if(.)
+		return FALSE
+
+	usr.set_interaction(src)
+
+	switch(href_list["operation"])
+		if("main")
+			state = STATE_DEFAULT
+
+		if("login")
+			var/mob/living/carbon/human/C = usr
+			var/obj/item/card/id/I = C.get_active_held_item()
+			if(istype(I))
+				if(check_access(I))
+					authenticated = 1
+				if(ACCESS_MARINE_CMO in I.access)
+					authenticated = 2
+					crew_announcement.announcer = GetNameAndAssignmentFromId(I)
+			else
+				I = C.wear_id
+				if(istype(I))
+					if(check_access(I))
+						authenticated = 1
+					if(ACCESS_MARINE_CMO in I.access)
+						authenticated = 2
+						crew_announcement.announcer = GetNameAndAssignmentFromId(I)
+		if("logout")
+			authenticated = 0
+			crew_announcement.announcer = ""
+
+		if("swipeidseclevel")
+			var/mob/M = usr
+			var/obj/item/card/id/I = M.get_active_held_item()
+			if(istype(I))
+				if(ACCESS_MARINE_CMO in I.access) //Let heads change the alert level.
+					switch(tmp_alertlevel)
+						if(-INFINITY to SEC_LEVEL_GREEN)
+							tmp_alertlevel = SEC_LEVEL_GREEN //Cannot go below green.
+						if(SEC_LEVEL_BLUE to SEC_LEVEL_BIO)
+							tmp_alertlevel = SEC_LEVEL_BLUE
+						if(SEC_LEVEL_BIO to INFINITY)
+							tmp_alertlevel = SEC_LEVEL_BIO //Can't go higher than Bio for the CMO
+
+					var/old_level = security_level
+					set_security_level(tmp_alertlevel)
+					if(security_level != old_level)
+						//Only notify the admins if an actual change happened
+						log_game("[key_name(usr)] has changed the security level to [get_security_level()].")
+						message_admins("[ADMIN_TPMONTY(usr)] has changed the security level to [get_security_level()].")
+						switch(security_level)
+							if(SEC_LEVEL_GREEN)
+								set_security_level(SEC_LEVEL_GREEN)
+								feedback_inc("alert_comms_green",1)
+							if(SEC_LEVEL_BLUE)
+								set_security_level(SEC_LEVEL_BLUE)
+								feedback_inc("alert_comms_blue",1)
+							if(SEC_LEVEL_BIO)
+								set_security_level(SEC_LEVEL_BIO)
+								feedback_inc("alert_comms_bio",1)
+				else
+					to_chat(usr, "<span class='warning'>You are not authorized to do this.</span>")
+				tmp_alertlevel = SEC_LEVEL_GREEN //Reset to green.
+				state = STATE_DEFAULT
+			else
+				to_chat(usr, "<span class='warning'>You need to swipe your ID.</span>")
+
+	/*	if("announce")
```

Comment:
Calling upper here makes commenting out things irrelevant. Also, big blocks of commented out code that shouldn't be there in the first place are bad.